### PR TITLE
Output residual sensitivities

### PR DESCRIPTION
**Pull request by roystgnr**
dR/dp should be useful for debugging.

### CODE REVIEW

File: src/solver/src/simulation.C [hunk 1]
```diff
@@ -309,6 +312,16 @@ namespace GRINS
     context.error_estimator = _error_estimator;
     context.print_qoi = _print_qoi;
 
+    if (_output_residual_sensitivities &&
+        !_forward_parameters.parameter_vector.size())
```
Comment (pbauman) [inline src/solver/src/simulation.C]: Should be useful for adjoint sensitivities too?

Comment (roystgnr) [inline src/solver/src/simulation.C]: Huh... yes, but not right away.  For some reason my code in libMesh is only creating permanent dR/dp vectors (well, retained until the user wants to get rid of them, anyway) in the forward case (and in some Hessian cases), but temporary vectors (deleted before the end of the calculation) in the adjoint case.  I'll need to refactor that before we can get at them.

**Comment by pbauman** [comment]
Cool. :+1:

**Comment by roystgnr** [comment]
Hmm... we're creating temporary dR/dp vectors in the adjoint sensitivity case because we sometimes need those vectors *unconstrained* to properly weight against heterogeneous BC adjoint solutions.  Need to think about this more and set up some better regression tests before refactoring, I think.

**Comment by pbauman** [comment]
Cool. Not urgent, IMO.

**Pull request merged**